File: c/2.c
// project euler problem #2
// http://anadoxin.org/blog
//
// $ gcc 2.c -o 2
// $ ./2

#include <stdio.h>

// slow :)
size_t getfib(size_t n) {
	if(n == 0 || n == 1)
		return 1;

	return getfib(n - 1) + getfib(n - 2);
}

int main() {
	size_t f, sum = 0, i = 0;

	do {
		f = getfib(++i);

		if(f % 2 == 0) {
			sum += f;
		}
	} while(f < 4000000);

	printf("%ld\n", sum);
	return 0;
}

File: pascal/2.pas
(*
   project euler problem #2
   http://anadoxin.org/blog

   $ fpc 2.pas
   $ ./2
*)

program euler2;

function getfib(n: UInt32): UInt32;
begin
    if (n = 0) or (n = 1) then
        getfib := 1
    else
        getfib := getfib(n - 1) + getfib(n - 2)
end;

var
    f, sum, i: UInt32;
begin
    i := 0;
    sum := 0;
    f := 0;

    while(f < 4000000) do
    begin
        i := i + 1;
        f := getfib(i);

        if (f mod 2 = 0) then
            sum := sum + f;
    end;

    writeln(sum);
end.
